File: yh_check/src/yh_check_sub.cpp
#include "ros/ros.h"
#include "yh_check/YhCheck.h"

bool distance = true;
bool camera = true;

void distanceCallback(const yh_check::YhCheck::ConstPtr& msg)
{
    distance = msg->data;
    if (distance && camera)
    {
        ROS_INFO("OK"); 
    }
}

void cameraCallback(const yh_check::YhCheck::ConstPtr& msg)
{
    camera = msg->data;
    if (distance && camera)
    {
        ROS_INFO("OK"); 
    }
}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "yh_check_sub"); 
    ros::NodeHandle nh;
    
    ros::Subscriber sub_distance = nh.subscribe("check_distance", 20, distanceCallback);
    ros::Subscriber sub_camera = nh.subscribe("check_camera", 20, cameraCallback); 

    ros::spin(); 

    return 0;
}

File: yh_check/scripts/yh_check_sub.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

import rospy
from yh_check.msg import YhCheck

class MyCheck:
    def __init__(self):
        self.sub_distance = rospy.Subscriber("check_distance", YhCheck, self.distanceCallback)
        self.sub_camera = rospy.Subscriber("check_camera", YhCheck, self.cameraCallback)
        self.sub_distance = True
        self.sub_camera = True

    def distanceCallback(self, msg):
        self.distance = msg.data
        if self.distance and self.camera:
            rospy.loginfo("OK")

    def cameraCallback(self, msg):
        self.camera = msg.data
        if self.distance and self.camera:
            rospy.loginfo("OK")

if __name__ == "__main__":
    rospy.init_node("yh_check_sub")
    my_check = MyCheck()
    rospy.spin()